src/main/java/seedu/address/model/entry/Date.java

```diff
@@ -0,0 +1,48 @@
+package seedu.address.model.entry;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+public class Date {
+    public static final String MESSAGE_CONSTRAINTS =
+            "Dates should be of the format dd-mm-yyyy and it should only contain numbers";
+    public static final String VALIDATION_REGEX = "^([0-2][0-9]||3[0-1])-(0[1-9]||[1-9]||1[0-2])-([0-9][0-9])?[0-9][0-9]$";
+    ;;
+    public final String date;

```

Comment:
Should we use `LocalDate` instead? Right now if I put 31-02-2022 it correctly adds the entry and does not return any error but it's an invalid date.

Reply:
Oh yea.. my bad the validation regex only checks for the validity of a string, can we change it after the base content has been merged?